build-logic/build-platform/build.gradle.kts

```diff
@@ -6,7 +8,8 @@ description = "Provides a platform that constrains versions of external dependen
 
 // Here you should declare versions which should be shared by the different modules of buildSrc itself
 val javaParserVersion = "3.18.0"
-val groovyVersion = "3.0.10"
+val groovyVersion = providers.gradleProperty("buildGroovyVersion")
+    .getOrElse(GroovySystem.getVersion())
```

Comment:
Regardless of changing the group: I don't think it makes sense to make this Kotlin version here configurable, since this is for the build logic in the Gradle build, and that should always use `GroovySystem.getVersion()`. We used `buildKotlinVersion` to use a newer Kotlin plugin to build our production Kotlin sources, though still against the old Kotlin library version. That is why I think `buildGroovyVersion` doesn't make any sense. Though you have been looking deeper into this, so maybe you have a use-case I don't see.

I'd say if you want to make the Groovy version we build into Gradle and which we test configurable, then you'd need to introduce make `ExternalModulesExtension.groovyVersion` configurable via a project property.

Reply:
OK.